vertical-pod-autoscaler/pkg/recommender/model/cluster_test.go

```diff
@@ -235,13 +235,15 @@ func TestMissingKeys(t *testing.T) {
 }
 
 func addVpa(cluster *ClusterState, id VpaID, annotations vpaAnnotationsMap, selector string) *Vpa {
-	var apiObject vpa_types.VerticalPodAutoscaler
-	apiObject.Namespace = id.Namespace
-	apiObject.Name = id.VpaName
-	apiObject.Annotations = annotations
+	apiObject := test.VerticalPodAutoscaler().WithNamespace(id.Namespace).
+		WithName(id.VpaName).WithContainer(testContainerID.ContainerName).WithAnnotations(annotations).Get()
```

Comment:
You've added WithContainer but I don't immediately see why. Could you please explain (here, not in code) so we can make sure this was on purpose?

Reply:
VPA builder panics if you call get before setting container name.